splitio/client/factory.py

```diff
@@ -528,22 +527,37 @@ def _build_localhost_factory(cfg):
     }
 
     synchronizers = SplitSynchronizers(
-        LocalSplitSynchronizer(cfg['splitFile'], storages['splits']),
-        None, None, None, None,
+        LocalSplitSynchronizer(cfg['splitFile'],
+                               storages['splits'],
+                               LocalhostMode.JSON if cfg['splitFile'][-5:].lower() == '.json' else LocalhostMode.LEGACY),
+        LocalSegmentSynchronizer(cfg['segmentDirectory'], storages['splits'], storages['segments']),
+        None, None, None,
     )
 
-    tasks = SplitTasks(
-        SplitSynchronizationTask(
+    split_sync_task = None
+    segment_sync_task = None
+    if cfg['localhostRefreshEnabled']:
```

Comment:
maybe you need to verified at this point if it's also JSON mode, I think this will be fired if customer enables with yaml

Reply:
will fix